codec-http/src/main/java/io/netty/handler/codec/http/QueryStringEncoder.java

```diff
@@ -245,6 +245,6 @@ private static char forDigit(int digit) {
      */
     private static boolean dontNeedEncoding(char ch) {
```

Comment:
Please also update the javadocs:

```
      unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
```

Reply:
Do I still need to add * in the javadoc?

Reply:
https://datatracker.ietf.org/doc/html/rfc3986#section-2.2 Here it says * is reserved. So I guess we need to encode it.

Reply:
Thanks

Reply:
@skyguard1 

After reading more, I'm leaning forward *NOT* to encode *. Sorry for the confusion, but encoding it or not may not be as straightforward as we thought. 


https://stackoverflow.com/questions/25085992/when-should-an-asterisk-be-encoded-in-an-http-url here it suggests not encoding *. 

From https://en.wikipedia.org/wiki/Percent-encoding#Percent_character
> In the "query" component of a URI (the part after a ? character), for example, / is still considered a reserved character but it normally has no reserved purpose, unless a particular URI scheme says otherwise. The character does not need to be percent-encoded when it has no reserved purpose.

I think it also applies to *.

Considering the adoption of encoding/decoding an asterisk listed in the above StackOverflow post, it shows only PHP rawurlencode() does the work. Java utils and JS utils do not encode. If we choose to encode an asterisk, we are at a risk to break our clients and downstreams.

Reply:
Can we separate adding `~` to this list and removing `*` from this list?

As mentioned in the original issue, the [spec seems clear](https://datatracker.ietf.org/doc/html/rfc3986#section-2.3) about not needing to encode `~` so that seem non-contentious.

Let us add `~` to this list in this PR and we can discuss removing `*` in a different PR.

Reply:
@NiteshKant Sure